service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -312,33 +312,63 @@ public void deleteGcpCloudContext(UUID workspaceId, AuthenticatedUserRequest use
         .submitAndWait(null);
   }
 
+  /**
+   * Retrieve the optional GCP cloud context NOTE: no user auth validation
```

Comment:
Seems risky to leave off the user validation. Can we at least make this more obvious in the name that this is by passing auth since other WorkspaceService methods do do auth?
I guess ditto all these other methods, or let's consider whether we should not be mixing auth checking with context retrieval in this class.

Reply:
OK, that is two votes for moving the GCP utility stuff elsewhere. I will pull it from here and the DAO and put it in its own service.